harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -263,85 +578,278 @@ def cleanup_iterator() -> None:
+                    batch = next(dataloader_iter)
+            except StopIteration:
+                return
+            yield batch
 
     def _run(self) -> None:
-        assert self.workloads is not None
-        for w, response_func in self.workloads:
+        if self.local_training:
             try:
-                if w.kind == workload.Workload.Kind.RUN_STEP:
-                    action = "training"
-                    metrics = self._train_for_step(
-                        w.step_id,
-                        w.num_batches,
-                        w.total_batches_processed,
-                    )
-                    response = {
-                        "metrics": metrics,
-                        "stop_requested": self.context.get_stop_requested(),
-                    }  # type: workload.Response
-                    metrics = self.context.distributed.broadcast(metrics)
-                    for callback in self.callbacks.values():
-                        callback.on_training_workload_end(
-                            avg_metrics=metrics["avg_metrics"],
-                            batch_metrics=metrics["batch_metrics"],
-                        )
-                    if (
-                        self.context.distributed.size > 1
-                        and not self.context._average_training_metrics
-                    ):
-                        warnings.warn(
-                            "Only the chief worker's training metrics are being reported, due "
-                            "to setting average_training_metrics to False.",
-                            UserWarning,
-                            stacklevel=2,
-                        )
-                elif w.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-                    action = "validation"
-                    response = {
-                        "metrics": self._compute_validation_metrics(),
-                        "stop_requested": self.context.get_stop_requested(),
-                    }
-                elif w.kind == workload.Workload.Kind.CHECKPOINT_MODEL:
-                    action = "checkpointing"
-                    uuid = ""
-                    if self.is_chief:
-                        metadata = {
-                            "determined_version": det.__version__,
-                            "steps_completed": self.steps_completed,
-                            "framework": f"torch-{torch.__version__}",
-                            "format": "pickle",
-                        }
-                        with self.context._core.checkpoint.store_path(metadata) as (
-                            path,
-                            storage_id,
-                        ):
-                            self._save(path)
-                            uuid = storage_id
-                        response = {"uuid": storage_id}
+                assert self.max_length, "max_length required for local training mode"
```

Comment:
if this checks user input (it does) it must be an if statement, not an assert statement.